// @looker/components/cjs/Link/Link.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.UnderlineTrue = exports.UnderlineFalse = exports.KitchenSink = exports.KeyColor = exports.External = exports.Basic = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _storybook = require("@looker/storybook");

var _Link = require("./Link");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var _default = {
  argTypes: _storybook.defaultArgTypes,
  component: _Link.Link,
  title: 'Link'
};
exports["default"] = _default;

var Template = function Template(args) {
  return _react["default"].createElement(_Link.Link, args, "A link to somewhere...");
};

var Basic = Template.bind({});
exports.Basic = Basic;
Basic.args = {
  href: 'https://google.com'
};
var KeyColor = Template.bind({});
exports.KeyColor = KeyColor;
KeyColor.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  keyColor: true
});
var External = Template.bind({});
exports.External = External;
External.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  isExternal: true
});
var KitchenSink = Template.bind({});
exports.KitchenSink = KitchenSink;
KitchenSink.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  isExternal: true,
  keyColor: true,
  underline: true
});
var UnderlineFalse = Template.bind({});
exports.UnderlineFalse = UnderlineFalse;
UnderlineFalse.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  underline: false
});
var UnderlineTrue = Template.bind({});
exports.UnderlineTrue = UnderlineTrue;
UnderlineTrue.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  underline: true
});
