// @looker/components/cjs/CopyToClipboard/CopyToClipboard.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LabelStyled = exports.ComponentStyled = exports.Basic = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _storybook = require("@looker/storybook");

var _Button = require("../Button/Button");

var _CopyToClipboard = require("./CopyToClipboard");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var Template = function Template(args) {
  return _react["default"].createElement(_CopyToClipboard.CopyToClipboard, args);
};

var Basic = Template.bind({});
exports.Basic = Basic;
Basic.args = {
  content: 'here is some text to be copied'
};
Basic.parameters = {
  storyshots: {
    disable: true
  }
};
var LabelStyled = Template.bind({});
exports.LabelStyled = LabelStyled;
LabelStyled.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  children: 'Copy Something',
  success: 'it was copied'
});
LabelStyled.parameters = {
  storyshots: {
    disable: true
  }
};
var ComponentStyled = Template.bind({});
exports.ComponentStyled = ComponentStyled;
ComponentStyled.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  children: _react["default"].createElement(_Button.Button, null, "Copy stuff"),
  success: _react["default"].createElement(_Button.Button, null, "Success")
});
ComponentStyled.parameters = {
  storyshots: {
    disable: true
  }
};
var _default = {
  argTypes: _storybook.defaultArgTypes,
  component: _CopyToClipboard.CopyToClipboard,
  title: 'CopyToClipboard'
};
exports["default"] = _default;
