// @looker/components/Calendar/MonthPicker.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
const _excluded = ["className", "date", "isTodaysMonth", "locale", "monthNumber", "onMonthClick", "selected", "selectedMonth", "style"],
      _excluded2 = ["callbacks"];

let _ = t => t,
    _t;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

import React, { useCallback } from 'react';
import styled from 'styled-components';
import pick from 'lodash/pick';
import { rippleHandlerKeys, rippleStyle, useBoundedRipple, useRippleHandlers } from '../Ripple';
import { isThisMonth } from './utils/dateConfirmations';
export const MonthPicker = styled(_ref => {
  var _locale$localize;

  let {
    className,
    date = new Date(),
    isTodaysMonth,
    locale,
    monthNumber,
    onMonthClick,
    selected,
    selectedMonth,
    style
  } = _ref,
      restProps = _objectWithoutProperties(_ref, _excluded);

  const _useBoundedRipple = useBoundedRipple({
    className,
    color: 'neutral',
    style
  }),
        {
    callbacks
  } = _useBoundedRipple,
        rippleProps = _objectWithoutProperties(_useBoundedRipple, _excluded2);

  const rippleHandlers = useRippleHandlers(callbacks, _objectSpread({}, pick(restProps, rippleHandlerKeys)), restProps.disabled);
  const isMonth = selectedMonth && isThisMonth(selectedMonth, monthNumber, date);
  isTodaysMonth = isTodaysMonth && isThisMonth(new Date(), monthNumber, date);
  return React.createElement("button", _extends({
    "aria-current": isTodaysMonth,
    "aria-selected": isMonth,
    type: "button",
    onClick: useCallback(() => onMonthClick(monthNumber), [monthNumber, onMonthClick])
  }, restProps, rippleProps, rippleHandlers), (_locale$localize = locale.localize) === null || _locale$localize === void 0 ? void 0 : _locale$localize.month(monthNumber, {
    width: 'abbreviated'
  }));
}).withConfig({
  displayName: "MonthPicker",
  componentId: "sc-mrnrtl-0"
})(_t || (_t = _`
  ${0}
  background: transparent;
  border: 1px solid ${0};
  border-radius: ${0};
  color: ${0};
  font-size: ${0};
  font-weight: ${0};
  height: ${0};
  width: ${0};
  &[aria-current='true'] {
    border: 1px solid ${0};
    color: ${0};
  }
  &[aria-selected='true'] {
    background: ${0};
    color: ${0};
  }
`), rippleStyle, ({
  theme
}) => theme.colors.ui2, ({
  theme
}) => theme.space.u5, ({
  theme
}) => theme.colors.text3, ({
  theme
}) => theme.fontSizes.xsmall, ({
  theme
}) => theme.fontWeights.medium, ({
  theme
}) => theme.space.u7, ({
  theme
}) => theme.space.u12, ({
  theme
}) => theme.colors.key, ({
  theme
}) => theme.colors.key, ({
  theme
}) => theme.colors.key, ({
  theme
}) => theme.colors.keyText);
