// @looker/components/cjs/Tooltip/Tooltip.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.RenderProp = exports.PlacementRight = exports.PlacementLeft = exports.PlacemenTop = exports.PerformanceTest = exports.OpenDelayNone = exports.Open = exports.NestedInPopover = exports.LargeTrigger = exports.DelayNone = exports.Basic = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _storybook = require("@looker/storybook");

var _Button = require("../Button");

var _Card = require("../Card");

var _Form = require("../Form");

var _Layout = require("../Layout");

var _Text = require("../Text");

var _Popover = require("../Popover");

var _2 = require("./");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var Template = function Template(args) {
  return _react["default"].createElement(_2.Tooltip, args, _react["default"].createElement(_Button.Button, {
    m: "xxxlarge"
  }, "Open Tooltip"));
};

var Basic = Template.bind({});
exports.Basic = Basic;
Basic.args = {
  content: 'Simple Content'
};
Basic.parameters = {
  storyshots: {
    disable: true
  }
};
var Open = Template.bind({});
exports.Open = Open;
Open.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  isOpen: true
});
Open.parameters = {
  storyshots: {
    disable: true
  }
};
var PlacemenTop = Template.bind({});
exports.PlacemenTop = PlacemenTop;
PlacemenTop.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  placement: 'top'
});
PlacemenTop.parameters = {
  storyshots: {
    disable: true
  }
};
var PlacementRight = Template.bind({});
exports.PlacementRight = PlacementRight;
PlacementRight.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  placement: 'right'
});
PlacementRight.parameters = {
  storyshots: {
    disable: true
  }
};
var PlacementLeft = Template.bind({});
exports.PlacementLeft = PlacementLeft;
PlacementLeft.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  content: 'Left',
  placement: 'left'
});
PlacementLeft.parameters = {
  storyshots: {
    disable: true
  }
};
var DelayNone = Template.bind({});
exports.DelayNone = DelayNone;
DelayNone.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  delay: 'none'
});
DelayNone.parameters = {
  storyshots: {
    disable: true
  }
};
var OpenDelayNone = Template.bind({});
exports.OpenDelayNone = OpenDelayNone;
OpenDelayNone.args = _objectSpread(_objectSpread({}, DelayNone.args), {}, {
  isOpen: true
});
OpenDelayNone.parameters = {
  storyshots: {
    disable: true
  }
};

var RenderProp = function RenderProp() {
  return _react["default"].createElement(_2.Tooltip, {
    content: "Start editing",
    placement: "right"
  }, function (tooltipProps) {
    return _react["default"].createElement(_Button.Button, (0, _extends2["default"])({
      m: "xxxlarge"
    }, tooltipProps), "Open Tooltip");
  });
};

exports.RenderProp = RenderProp;
RenderProp.parameters = {
  storyshots: {
    disable: true
  }
};

var LargeTrigger = function LargeTrigger() {
  return _react["default"].createElement(_2.Tooltip, {
    content: "See what happens when you scroll",
    placement: "right"
  }, _react["default"].createElement(_Card.Card, {
    width: 500,
    height: 800,
    raised: true,
    p: "u5"
  }, "Very large trigger"));
};

exports.LargeTrigger = LargeTrigger;
LargeTrigger.parameters = {
  storyshots: {
    disable: true
  }
};

var NestedInPopover = function NestedInPopover() {
  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      prevent = _useState2[0],
      setPrevent = _useState2[1];

  function handleChange(e) {
    setPrevent(e.currentTarget.checked);
  }

  var _useState3 = (0, _react.useState)('N/A'),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      lastEvent = _useState4[0],
      setLastEvent = _useState4[1];

  var getHandler = function getHandler(text) {
    return function (e) {
      setLastEvent(text);

      if (prevent) {
        e.preventDefault();
      }
    };
  };

  var handlers = {
    onBlur: getHandler('blur'),
    onClick: getHandler('click'),
    onFocus: getHandler('focus'),
    onMouseOut: getHandler('mouse out'),
    onMouseOver: getHandler('mouse over')
  };
  return _react["default"].createElement(_Layout.SpaceVertical, {
    p: "u5"
  }, _react["default"].createElement(_Text.Text, null, "Last event: ", lastEvent), _react["default"].createElement(_Layout.Space, null, _react["default"].createElement(_Popover.Popover, {
    content: _react["default"].createElement(_Popover.PopoverContent, null, "Some content")
  }, _react["default"].createElement(_2.Tooltip, {
    content: "Some tooltip"
  }, _react["default"].createElement(_Button.Button, handlers, "Open"))), _react["default"].createElement(_Form.FieldToggleSwitch, {
    on: prevent,
    onChange: handleChange,
    label: "Prevent default"
  })));
};

exports.NestedInPopover = NestedInPopover;
NestedInPopover.parameters = {
  storyshots: {
    disable: true
  }
};

var PerformanceTest = function PerformanceTest() {
  var _useState5 = (0, _react.useState)(''),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      value = _useState6[0],
      setValue = _useState6[1];

  var handleChange = function handleChange(e) {
    return setValue(e.currentTarget.value);
  };

  var lastRenderRef = (0, _react.useRef)(Date.now());
  (0, _react.useEffect)(function () {
    var now = Date.now();
    var diff = now - lastRenderRef.current;
    console.log(diff);
    lastRenderRef.current = now;
  });
  return _react["default"].createElement("div", null, _react["default"].createElement("p", null, "Type fast then hold down delete:"), _react["default"].createElement("input", {
    type: "text",
    value: value,
    onChange: handleChange
  }), _react["default"].createElement("p", null, "The text shouldn't freeze due to main thread being blocked."), _react["default"].createElement("div", null, Array.from(Array(1000), function (_, i) {
    return _react["default"].createElement(_2.Tooltip, {
      key: i,
      content: "I'm a tooltip"
    }, _react["default"].createElement("button", null, "Hover me"));
  })));
};

exports.PerformanceTest = PerformanceTest;
PerformanceTest.parameters = {
  storyshots: {
    disable: true
  }
};
var _default = {
  argTypes: _storybook.defaultArgTypes,
  component: _2.Tooltip,
  title: 'Tooltip'
};
exports["default"] = _default;
