// @looker/components/cjs/Calendar/Calendar.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Range = exports.Basic = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _i18n = require("@looker/i18n");

var _react = _interopRequireWildcard(require("react"));

var _storybook = require("@looker/storybook");

var _Calendar = require("./Calendar");

var _excluded = ["onSelectDate", "selectedDate", "viewMonth"],
    _excluded2 = ["onSelectRange", "selectedRange", "viewMonth"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var _default = {
  argTypes: _objectSpread(_objectSpread({}, _storybook.defaultArgTypes), {}, {
    locale: {
      control: {
        type: 'select'
      },
      mapping: _i18n.dateFnLocaleMap,
      options: Object.keys(_i18n.dateFnLocaleMap)
    }
  }),
  component: _Calendar.Calendar,
  title: 'Calendar'
};
exports["default"] = _default;

var Template = function Template(_ref) {
  var onSelectDate = _ref.onSelectDate,
      selectedDate = _ref.selectedDate,
      viewMonthProp = _ref.viewMonth,
      args = (0, _objectWithoutProperties2["default"])(_ref, _excluded);

  var _useState = (0, _react.useState)(selectedDate),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      date = _useState2[0],
      setDate = _useState2[1];

  var handleSelect = function handleSelect(newDate) {
    onSelectDate === null || onSelectDate === void 0 ? void 0 : onSelectDate(newDate);
    setDate(newDate);
  };

  var _useState3 = (0, _react.useState)(viewMonthProp),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      viewMonth = _useState4[0],
      setViewMonth = _useState4[1];

  return _react["default"].createElement(_Calendar.Calendar, (0, _extends2["default"])({}, args, {
    onSelectDate: handleSelect,
    selectedDate: date,
    viewMonth: viewMonth,
    onMonthChange: setViewMonth
  }));
};

var TemplateRange = function TemplateRange(_ref2) {
  var onSelectRange = _ref2.onSelectRange,
      selectedRange = _ref2.selectedRange,
      viewMonthProp = _ref2.viewMonth,
      args = (0, _objectWithoutProperties2["default"])(_ref2, _excluded2);

  var _useState5 = (0, _react.useState)(selectedRange),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      range = _useState6[0],
      setRange = _useState6[1];

  var handleSelect = function handleSelect(newRange) {
    onSelectRange === null || onSelectRange === void 0 ? void 0 : onSelectRange(newRange);
    setRange(newRange);
  };

  var _useState7 = (0, _react.useState)(viewMonthProp),
      _useState8 = (0, _slicedToArray2["default"])(_useState7, 2),
      viewMonth = _useState8[0],
      setViewMonth = _useState8[1];

  return _react["default"].createElement(_Calendar.Calendar, (0, _extends2["default"])({}, args, {
    isRange: true,
    onSelectRange: handleSelect,
    selectedRange: range,
    viewMonth: viewMonth,
    onMonthChange: setViewMonth
  }));
};

var Basic = Template.bind({});
exports.Basic = Basic;
Basic.args = {
  viewMonth: new Date('Jul 1, 2021')
};
Basic.parameters = {
  storyshots: {
    disable: true
  }
};
var Range = TemplateRange.bind({});
exports.Range = Range;
Range.args = {
  viewMonth: new Date('Feb 1, 2022')
};
Range.parameters = {
  storyshots: {
    disable: true
  }
};
