// @looker/components/CopyToClipboard/CopyToClipboard.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

import React from 'react';
import { defaultArgTypes as argTypes } from '@looker/storybook';
import { Button } from '../Button/Button';
import { CopyToClipboard } from './CopyToClipboard';

const Template = args => React.createElement(CopyToClipboard, args);

export const Basic = Template.bind({});
Basic.args = {
  content: 'here is some text to be copied'
};
Basic.parameters = {
  storyshots: {
    disable: true
  }
};
export const LabelStyled = Template.bind({});
LabelStyled.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  children: 'Copy Something',
  success: 'it was copied'
});
LabelStyled.parameters = {
  storyshots: {
    disable: true
  }
};
export const ComponentStyled = Template.bind({});
ComponentStyled.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  children: React.createElement(Button, null, "Copy stuff"),
  success: React.createElement(Button, null, "Success")
});
ComponentStyled.parameters = {
  storyshots: {
    disable: true
  }
};
export default {
  argTypes,
  component: CopyToClipboard,
  title: 'CopyToClipboard'
};
