// @looker/components/Accordion2/AccordionIndicator.js
import _extends from "@babel/runtime/helpers/extends";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
const _excluded = ["children", "density", "indicatorPosition"];

let _ = t => t,
    _t,
    _t2;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

import pick from 'lodash/pick';
import React from 'react';
import styled from 'styled-components';
import { StyledIconBase } from '@styled-icons/styled-icon';
import { rippleHandlerKeys, rippleStyle, useRipple, useRippleHandlers } from '../Ripple';
import { accordionDimensions } from './accordionDimensions';

const size = (density = 0) => accordionDimensions(density).indicatorSize;

const gap = (density = 0) => accordionDimensions(density).indicatorGap;

export const AccordionIndicator = styled(_ref => {
  let {
    children,
    density,
    indicatorPosition
  } = _ref,
      props = _objectWithoutProperties(_ref, _excluded);

  const {
    callbacks,
    className: rippleClassName,
    style: rippleStyle
  } = useRipple({
    color: 'neutral'
  });
  const rippleHandlers = useRippleHandlers(callbacks, _objectSpread({}, pick(_objectSpread({}, props), rippleHandlerKeys)));
  const rippleContainerProps = {
    className: rippleClassName,
    style: rippleStyle
  };
  const isIndicatorToggleOnly = props.tabIndex === -1;
  return React.createElement("div", _extends({}, props, isIndicatorToggleOnly && rippleHandlers), React.createElement(RippleContainer, _extends({
    density: density || 0
  }, rippleContainerProps), children));
}).withConfig({
  displayName: "AccordionIndicator",
  componentId: "sc-1w66fqe-0"
})(_t || (_t = _`
  align-items: center;
  display: flex;
  justify-content: center;
  outline: none;

  ${0}

  ${0} {
    height: ${0};
    /*
      Default vertical-align is set to middle which shifts indicator icon
      below mid-point
    */
    vertical-align: baseline;
    width: ${0};
  }
`), ({
  density,
  indicatorPosition,
  theme: {
    space
  }
}) => indicatorPosition === 'left' ? `margin-right: ${space[gap(density)]};` : `margin-left: ${space[gap(density)]};`, StyledIconBase, ({
  density,
  theme
}) => theme.sizes[size(density)], ({
  density,
  theme
}) => theme.sizes[size(density)]);
const RippleContainer = styled.div.withConfig({
  displayName: "AccordionIndicator__RippleContainer",
  componentId: "sc-1w66fqe-1"
})(_t2 || (_t2 = _`
  ${0}
  border-radius: 50%;
  height: ${0};
  width: ${0};
`), rippleStyle, ({
  density,
  theme
}) => theme.sizes[size(density)], ({
  density,
  theme
}) => theme.sizes[size(density)]);
