// @looker/components/Layout/Space/SpaceVertical.js
let _ = t => t,
    _t,
    _t2;

import { shouldForwardProp } from '@looker/design-tokens';
import styled, { css } from 'styled-components';
import { defaultGap, spaceCSS } from './Space';

const flexGap = ({
  gap: _gap = defaultGap,
  reverse
}) => css(_t || (_t = _`
  @supports (-moz-appearance: none) {
    gap: ${0};
  }

  @supports not (-moz-appearance: none) {
    && > * {
      margin-top: ${0};
    }

    ${0}
  }
`), ({
  theme: {
    space
  }
}) => space[_gap], ({
  theme: {
    space
  }
}) => space[_gap], ({
  theme: {
    space
  }
}) => reverse ? `&& > *:last-child { margin-top: ${space.none}; }` : `&& > *:first-child { margin-top: ${space.none}; }`);

export const SpaceVertical = styled.div.withConfig({
  shouldForwardProp,
  displayName: "SpaceVertical",
  componentId: "sc-tnj0rp-0"
}).attrs(({
  align: _align = 'flex-start',
  width: _width = '100%'
}) => {
  if (['start', 'end'].includes('align')) {
    _align = `flex-${_align}`;
  }

  return {
    alignItems: _align,
    width: _width
  };
})(_t2 || (_t2 = _`
  ${0}
  ${0}
  flex-direction: ${0};
`), spaceCSS, flexGap, ({
  reverse
}) => reverse ? 'column-reverse' : 'column');
