// @looker/components/Link/Link.stories.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

import React from 'react';
import { defaultArgTypes as argTypes } from '@looker/storybook';
import { Link } from './Link';
export default {
  argTypes,
  component: Link,
  title: 'Link'
};

const Template = args => React.createElement(Link, args, "A link to somewhere...");

export const Basic = Template.bind({});
Basic.args = {
  href: 'https://google.com'
};
export const KeyColor = Template.bind({});
KeyColor.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  keyColor: true
});
export const External = Template.bind({});
External.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  isExternal: true
});
export const KitchenSink = Template.bind({});
KitchenSink.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  isExternal: true,
  keyColor: true,
  underline: true
});
export const UnderlineFalse = Template.bind({});
UnderlineFalse.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  underline: false
});
export const UnderlineTrue = Template.bind({});
UnderlineTrue.args = _objectSpread(_objectSpread({}, Basic.args), {}, {
  underline: true
});
