// @looker/components/cjs/Layout/Space/Space.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.spaceCSS = exports.defaultGap = exports.Space = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _designTokens = require("@looker/design-tokens");

var _common = require("../utils/common");

var _templateObject, _templateObject2, _templateObject3, _templateObject4;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var defaultGap = 'u4';
exports.defaultGap = defaultGap;
var spaceCSS = (0, _styledComponents.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n  ", "\n\n  display: flex;\n"])), _common.commonLayoutCSS, _designTokens.flexbox);
exports.spaceCSS = spaceCSS;

var fauxGap = function fauxGap(_ref) {
  var _ref$gap = _ref.gap,
      gap = _ref$gap === void 0 ? defaultGap : _ref$gap,
      reverse = _ref.reverse;
  return (0, _styledComponents.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  && > * {\n    margin-right: ", ";\n  }\n\n  ", "\n"])), function (_ref2) {
    var space = _ref2.theme.space;
    return space[gap];
  }, function (_ref3) {
    var space = _ref3.theme.space;
    return reverse ? "&& > *:first-child { margin-right: ".concat(space.none, "; }") : "&& > *:last-child { margin-right: ".concat(space.none, "; }");
  });
};

var flexGap = function flexGap(_ref4) {
  var _ref4$gap = _ref4.gap,
      gap = _ref4$gap === void 0 ? defaultGap : _ref4$gap,
      reverse = _ref4.reverse;
  return (0, _styledComponents.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n  @supports (-moz-appearance: none) {\n    gap: 0 ", ";\n  }\n\n  @supports not (-moz-appearance: none) {\n    ", "\n  }\n"])), function (_ref5) {
    var space = _ref5.theme.space;
    return space[gap];
  }, fauxGap({
    gap: gap,
    reverse: reverse
  }));
};

var verticalAlign = (0, _designTokens.variant)({
  prop: 'align',
  variants: {
    center: {
      alignItems: 'center'
    },
    end: {
      alignItems: 'flex-end'
    },
    start: {
      alignItems: 'flex-start'
    }
  }
});
var justify = (0, _designTokens.variant)({
  prop: 'justify',
  variants: {
    center: {
      justifyContent: 'center'
    },
    end: {
      justifyContent: 'flex-end'
    },
    start: {
      justifyContent: 'flex-start'
    }
  }
});

var Space = _styledComponents["default"].div.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp,
  displayName: "Space",
  componentId: "sc-zsz5hl-0"
}).attrs(function (_ref6) {
  var _ref6$alignItems = _ref6.alignItems,
      alignItems = _ref6$alignItems === void 0 ? 'center' : _ref6$alignItems,
      _ref6$width = _ref6.width,
      width = _ref6$width === void 0 ? '100%' : _ref6$width;
  return {
    alignItems: alignItems,
    width: width
  };
})(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n  ", "\n  ", "\n  flex-direction: ", ";\n\n  ", "\n  ", "\n  ", "\n  ", "\n"])), spaceCSS, function (_ref7) {
  var stretch = _ref7.stretch;
  return !stretch && verticalAlign;
}, function (_ref8) {
  var stretch = _ref8.stretch;
  return !stretch && justify;
}, function (_ref9) {
  var reverse = _ref9.reverse;
  return reverse ? 'row-reverse' : 'row';
}, function (_ref10) {
  var around = _ref10.around;
  return around && 'justify-content: space-around;';
}, function (_ref11) {
  var between = _ref11.between;
  return between && 'justify-content: space-between;';
}, function (_ref12) {
  var evenly = _ref12.evenly;
  return evenly && 'justify-content: space-evenly;';
}, function (_ref13) {
  var around = _ref13.around,
      between = _ref13.between,
      evenly = _ref13.evenly;
  return !around && !between && !evenly && flexGap;
});

exports.Space = Space;
