// @looker/components/cjs/Layout/Space/SpaceVertical.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SpaceVertical = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _designTokens = require("@looker/design-tokens");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Space = require("./Space");

var _templateObject, _templateObject2;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var flexGap = function flexGap(_ref) {
  var _ref$gap = _ref.gap,
      gap = _ref$gap === void 0 ? _Space.defaultGap : _ref$gap,
      reverse = _ref.reverse;
  return (0, _styledComponents.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  @supports (-moz-appearance: none) {\n    gap: ", ";\n  }\n\n  @supports not (-moz-appearance: none) {\n    && > * {\n      margin-top: ", ";\n    }\n\n    ", "\n  }\n"])), function (_ref2) {
    var space = _ref2.theme.space;
    return space[gap];
  }, function (_ref3) {
    var space = _ref3.theme.space;
    return space[gap];
  }, function (_ref4) {
    var space = _ref4.theme.space;
    return reverse ? "&& > *:last-child { margin-top: ".concat(space.none, "; }") : "&& > *:first-child { margin-top: ".concat(space.none, "; }");
  });
};

var SpaceVertical = _styledComponents["default"].div.withConfig({
  shouldForwardProp: _designTokens.shouldForwardProp,
  displayName: "SpaceVertical",
  componentId: "sc-tnj0rp-0"
}).attrs(function (_ref5) {
  var _ref5$align = _ref5.align,
      align = _ref5$align === void 0 ? 'flex-start' : _ref5$align,
      _ref5$width = _ref5.width,
      width = _ref5$width === void 0 ? '100%' : _ref5$width;

  if (['start', 'end'].includes('align')) {
    align = "flex-".concat(align);
  }

  return {
    alignItems: align,
    width: width
  };
})(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n  ", "\n  flex-direction: ", ";\n"])), _Space.spaceCSS, flexGap, function (_ref6) {
  var reverse = _ref6.reverse;
  return reverse ? 'column-reverse' : 'column';
});

exports.SpaceVertical = SpaceVertical;
